src/Renderer/ThreeExtended/PlanarControls.js

```diff
@@ -243,53 +252,58 @@ function PlanarControls(view, options = {}) {
     * The rotate movement is previously initiated in initiateRotation()
     * Compute the new position value and update the camera controls.
     */
-    this.handleRotation = function handleRotation() {
-        // angle deltas
-        // deltaMousePosition is computed in onMouseMove / onMouseDown s
-        const thetaDelta = -this.rotateSpeed * this.deltaMousePosition.x / window.innerWidth;
-        const phiDelta = -this.rotateSpeed * this.deltaMousePosition.y / window.innerHeight;
-
-        // the vector from centerPoint (focus point) to camera position
-        const offset = this.camera.position.clone().sub(this.centerPoint);
-
-        const quat = new THREE.Quaternion().setFromUnitVectors(this.camera.up, new THREE.Vector3(0, 0, 1));
-        const quatInverse = quat.clone().inverse();
-
-        if (thetaDelta !== 0 || phiDelta !== 0) {
-            if ((this.phi + phiDelta >= this.minZenithAngle)
-            && (this.phi + phiDelta <= this.maxZenithAngle)
-            && phiDelta !== 0) {
-                // rotation around X (altitude)
-                this.phi += phiDelta;
-                offset.applyQuaternion(quat);
-
-                const rotationXQuaternion = new THREE.Quaternion();
-                const vector = new THREE.Vector3();
-
-                vector.setFromMatrixColumn(this.camera.matrix, 0);
-                rotationXQuaternion.setFromAxisAngle(vector, phiDelta);
-                offset.applyQuaternion(rotationXQuaternion);
-                offset.applyQuaternion(quatInverse);
-            }
-            if (thetaDelta !== 0) {
-                // rotation around Z (azimuth)
-
-                const rotationZQuaternion = new THREE.Quaternion();
-                rotationZQuaternion.setFromAxisAngle(new THREE.Vector3(0, 0, 1), thetaDelta);
-                offset.applyQuaternion(rotationZQuaternion);
+    this.handleRotation = (() => {
+        const quat = new THREE.Quaternion();
+        let quatInverse = new THREE.Quaternion();
+
+        return () => {
+            // angle deltas
+            // deltaMousePosition is computed in onMouseMove / onMouseDown s
+            const thetaDelta = -this.rotateSpeed * deltaMousePosition.x / this.domElement.clientWidth;
+            const phiDelta = -this.rotateSpeed * deltaMousePosition.y / this.domElement.clientHeight;
+
+            // the vector from centerPoint (focus point) to camera position
+            const offset = this.camera.position.clone().sub(centerPoint);
+
+            quat.setFromUnitVectors(this.camera.up, new THREE.Vector3(0, 0, 1));
+            quatInverse = quat.clone().inverse();
```

Comment:
what about `quat.setFromUnitVectors(this.camera.up, new THREE.Vector3(0, 0, 1)).inverse()` and using `quat` instead of quatInverse below? It saves a `clone()`.